Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/Edge2QuadTopologicalMapping.cpp

```diff
@@ -97,6 +97,53 @@ void Edge2QuadTopologicalMapping::init()
     }
 
     unsigned int N = d_nbPointsOnEachCircle.getValue();
+    bool modelsOk = true;
+
+    // Check input topology
+    if (!fromModel)
+    {
+        // If the input topology link isn't set by the user, the TopologicalMapping::create method tries to find it.
+        // If it is null at this point, it means no input mesh topology could be found.
+        msg_error() << "No input mesh topology found. Consider setting the '" << fromModel.getName() << "' data attribute.";
+        modelsOk = false;
+    }
+    else
+    {
+        // Making sure the input topology is derived from the edge topology container
+        if (!dynamic_cast<container::dynamic::EdgeSetTopologyContainer *>(fromModel.get()))
+        {
+            msg_error() << "The input topology '" << fromModel.getPath() << "' is not a derived class of EdgeSetTopologyContainer. "
```

Comment:
I am concerned about the term "derived class", which is a programming term whereas you talk to a user that may not be familiar with this term.

You can also improve the message by giving the list of available classes:
```cpp
sofa::core::ObjectFactory::getInstance()->listClassesDerivedFrom<container::dynamic::EdgeSetTopologyContainer>()
```

Reply:
Very nice!